sktime/forecasting/compose/_by_group_forecaster.py

```diff
@@ -0,0 +1,526 @@
+            if self.get_tag("scitype:y")=="univariate":
+                guaranteed to have a single column/variable
+            if self.get_tag("scitype:y")=="multivariate":
+                guaranteed to have 2 or more columns
+            if self.get_tag("scitype:y")=="both": no restrictions apply
+        X :  sktime time series object, optional (default=None)
+            guaranteed to be of an mtype in self.get_tag("X_inner_mtype")
+            Exogeneous time series for the forecast
+        update_params : bool, optional (default=True)
+            whether model parameters should be updated
+
+        Returns
+        -------
+        self : reference to self
+        """
+        self.chosen_forecaster_.update(y=y, X=X, update_params=update_params)
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for forecasters.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        param1 = {
+            "forecasters": {
+                "smooth": NaiveForecaster(),
+                "erratic": PolynomialTrendForecaster(),
+                "intermittent": Croston(),
+                "lumpy": NaiveForecaster(),
+            },
+            "transformer": ADICVTransformer(features=["class"]),
+            "fallback_forecaster": None,
+        }
+
+        # Attempting to utilize the fallback forecaster
+        param2 = {
+            "forecasters": {},
+            "transformer": ADICVTransformer(features=["class"]),
+            "fallback_forecaster": Croston(),
+        }
+
+        params = [param1, param2]
+        return params
+
+    @property
+    def _forecasters(self):
+        """Provides an internal list of the forecasters available.
+
+        Each list item is a tuple of the format (category, forecaster)
+        where the category for which the respective forecaster is chosen
+        and the forecaster itself as the values for each tuple.
+
+        Returns
+        -------
+        forecasters : list[tuple[str, strsktime forecasters]]
+            The list of forecasters which is returned. Also includes the
+            fallback forecaster with the category: "fallback_forecaster"
+        """
+        return list(self.forecasters.values()) + [
```

Comment:
should this be `items` rather than `values`?

Reply:
Oh, that's right! I'll make the change

Reply:
I find it odd that the tests did not detect this, how about adding an extra test that checks that the `get_params` keys are as expected?

Further, I think you may want to check whether you have to coerce the keys to `str`, or whether it works without giving trouble (because this happens somewhere else already).

Reply:
Yes, I'm adding in that check with the next commit. Could you explain what you mean by coercing the keys to string? Is that in reference to the `"fallback_forecaster", self.fallback_forecaster` line where I specify fallback forecaster as a string key?

Reply:
If you run `get_params` with `deep=True`, nested parameter keys will be strings `component__parametername`.

For the `_HeterogeneousMetaEstimator` to work, the name in name/estimator tuples needs to be string.

In cases where the transformer outputs non-string values (e.g., integers), in the property you may have to coerce the integers to strings - or maybe it is handled already somewhere.